governance/xc_admin/packages/xc_admin_common/src/contracts/EvmPythUpgradable.ts

```diff
@@ -0,0 +1,98 @@
+import {
+  Contract,
+  ContractType,
+  NetworkId,
+  SendGovernanceInstruction,
+  SyncOp,
+  WormholeAddress,
+  WormholeNetwork,
+} from "./Contract";
+import {
+  ChainId,
+  SetValidPeriodInstruction,
+} from "@pythnetwork/xc-governance-sdk";
+import { ethers } from "ethers";
+
+export class EvmPythUpgradable implements Contract<EvmPythUpgradableState> {
+  public type = ContractType.EvmPythUpgradable;
+  public networkId;
+  private address;
+
+  private contract: ethers.Contract;
+
+  constructor(
+    networkId: NetworkId,
+    address: string,
+    contract: ethers.Contract
+  ) {
+    this.networkId = networkId;
+    this.address = address;
+    this.contract = contract;
+  }
+
+  public getAddress() {
+    return this.address;
+  }
+
+  // TODO: these getters will need the full PythUpgradable ABI
+  public async getAuthority(): Promise<WormholeAddress> {
+    // FIXME: read from data sources
+    return {
+      emitter: "123454",
+      chainId: 1,
+      network: "mainnet",
+    };
+  }
+
+  // get the chainId that identifies this contract
+  public async getChainId(): Promise<ChainId> {
+    // FIXME: read from data sources
+    return 23;
+  }
+
+  public async getState(): Promise<EvmPythUpgradableState> {
+    const bytecodeSha = ethers.utils.sha256(
+      (await this.contract.provider.getCode(this.contract.address)) as string
+    );
+    const validTimePeriod =
+      (await this.contract.getValidTimePeriod()) as bigint;
+    return {
+      bytecodeSha,
+      validTimePeriod: validTimePeriod.toString(),
+    };
+  }
+
+  public async sync(target: EvmPythUpgradableState): Promise<SyncOp[]> {
+    const myState = await this.getState();
+    const authority = await this.getAuthority();
+    const myChainId = await this.getChainId();
+    const whInstructions = [];
+
+    if (myState.validTimePeriod !== target.validTimePeriod) {
+      whInstructions.push(
+        new SetValidPeriodInstruction(myChainId, BigInt(target.validTimePeriod))
+      );
+    }
+
+    return whInstructions.map(
+      (value) =>
+        new SendGovernanceInstruction(
+          value,
+          authority,
+          this.submitGovernanceVaa
+        )
+    );
+  }
+
+  public async submitGovernanceVaa(vaa: string): Promise<boolean> {
+    // FIXME: also needs the full PythUpgradable ABI
```

Comment:
You will just need to single abi to executeGovernanceInstruction. You can manually write that down and it's fine. I think full abi is not needed at all.